### ISSUE TITLE
Improve pathing of radial top/bottom infill patterns so they work monotonically instead of randomly.

### ISSUE THREAD

**Issue by slothy89** [SoftFever/OrcaSlicer#6670]
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

When using the Archimedean Chords and Octogram Spiral infills for top and bottom surfaces, the slicer will nicely spiral the infill from/to the center where it can form a continuous line, but as soon as the pattern intersects a wall, it randomly fills the remaining area leaving blemishes in the final finish. This is mostly problematic when using as a top surface infill pattern. But also noticable when printing with coextruded filaments like dual and tri-colour Silk PLAs on the bottom layer as well.
![IMG_0573](https://github.com/user-attachments/assets/9483af92-96cf-4669-9789-dffb87acfc4e)
![image](https://github.com/user-attachments/assets/aec56fbd-0b77-44fd-9200-da2f57e3254f)

This is easy to reproduce by creating a Primitive object with a high aspect rectangular shape, and setting the top layer fill to Archimedean Chords or Octogram Spiral.

### Which printers will be beneficial to this feature?

All

### Describe the solution you'd like

When the pattern intersects a wall it should expand the remaining infill in a similar manner to Monotonic infill, where it always ensures new lines continue from the last printed area. This difference is similar to how Monotonic vs Rectilinear behave.

Doing this will prevent uneven surfaces forming in the finish and leaving a clean pattern for decorative purposes. Which is the entire point of these infill types.

### Describe alternatives you've considered

Concentric tends to work well at maintaining a more monotonic fill path, but even that too sometimes exhibits this behaviour.

**Comment by zjohnson16** [comment]
+1

**Comment by jakuaari** [comment]
I noticed this recently and it ruined a print for me. I did not think to check what the order for the printed lines would be, instead assuming it would do them in at least some sensible order. I guess this pattern could be called "the worst possible surface finish" in the slicer instead. The print order makes absolutely no sense, skipping back and forther even adjacent lines.

![Image](https://github.com/user-attachments/assets/e76c45cb-dd70-48e5-8c68-a5b030f85458)
